pkg/usersettings/handlers.go

```diff
@@ -0,0 +1,340 @@
+
+	return &currentUser, nil
+}
+
+func (h *UserSettingsHandler) createRole(role *Role) (*http.Response, error) {
+	path := fmt.Sprintf("/apis/rbac.authorization.k8s.io/v1/namespaces/%s/roles", role.Metadata.Namespace)
+	return h.sendRequest(http.MethodPost, path, role)
+}
+
+func (h *UserSettingsHandler) createRoleBinding(roleBinding *RoleBinding) (*http.Response, error) {
+	path := fmt.Sprintf("/apis/rbac.authorization.k8s.io/v1/namespaces/%s/rolebindings", roleBinding.Metadata.Namespace)
+	return h.sendRequest(http.MethodPost, path, roleBinding)
+}
+
+func (h *UserSettingsHandler) createConfigMap(configMap *ConfigMap) (*http.Response, error) {
+	path := fmt.Sprintf("/api/v1/namespaces/%s/configmaps", configMap.Metadata.Namespace)
+	return h.sendRequest(http.MethodPost, path, configMap)
+}
+
+func (h *UserSettingsHandler) getConfigMap(namespace string, name string) (*http.Response, error) {
+	path := fmt.Sprintf("/api/v1/namespaces/%s/configmaps/%s", namespace, name)
+	return h.sendRequest(http.MethodGet, path, nil)
+}
+
+func (h *UserSettingsHandler) deleteConfigMap(namespace string, name string) (*http.Response, error) {
+	path := fmt.Sprintf("/api/v1/namespaces/%s/configmaps/%s", namespace, name)
+	return h.sendRequest(http.MethodDelete, path, nil)
+}
+
+func (h *UserSettingsHandler) deleteRole(namespace string, name string) (*http.Response, error) {
+	path := fmt.Sprintf("/apis/rbac.authorization.k8s.io/v1/namespaces/%s/roles/%s", namespace, name)
+	return h.sendRequest(http.MethodDelete, path, nil)
+}
+
+func (h *UserSettingsHandler) deleteRoleBinding(namespace string, name string) (*http.Response, error) {
+	path := fmt.Sprintf("/apis/rbac.authorization.k8s.io/v1/namespaces/%s/rolebindings/%s", namespace, name)
+	return h.sendRequest(http.MethodDelete, path, nil)
+}
+
+func (h *UserSettingsHandler) sendRequest(method string, path string, content interface{}) (*http.Response, error) {
+	url := proxy.SingleJoiningSlash(h.Endpoint, path)
+
+	body, err := json.Marshal(content)
+	if err != nil {
+		return nil, err
+	}
+
+	fmt.Printf("sendRequest %s %s\n%s\n", method, path, string(body))
+
+	req, err := http.NewRequest(method, url, bytes.NewReader(body))
+	if err != nil {
+		return nil, err
+	}
+
+	req.Header.Set("Authorization", fmt.Sprintf("Bearer %s", h.ServiceAccountToken))
+	req.Header.Set("Accept", "application/json")
+	if len(body) > 0 {
+		req.Header.Set("Content-Type", "application/json")
+		req.Header.Set("Content-Length", fmt.Sprintf("%d", len(body)))
+	}
+
+	response, err := h.Client.Do(req)
+	if err != nil {
+		return nil, err
+	}
+
+	fmt.Printf("sendRequest %d %s\n", response.StatusCode, response.Status)
+
+	return response, nil
+}
+
+func (h *UserSettingsHandler) copy(receivedResponse *http.Response, w http.ResponseWriter) {
+	w.Header().Set("Content-Type", receivedResponse.Header.Get("Content-Type"))
+	w.Header().Set("Content-Length", receivedResponse.Header.Get("Content-Length"))
+	w.WriteHeader(receivedResponse.StatusCode)
```

Comment:
At the very least, we need `Content-Encoding` as well if present. It's probably a good idea to add cache control headers. We should make sure the security-related headers are coming through as well.

Reply:
Should I pass all headers from the k8 response to the bridge response? I skipped this because that are more lines of code, but I think this make sense then.

Reply:
It's hard to make a recommendation since we're sort of reimplementing the proxy here. These are the headers we filter from response:

https://github.com/openshift/console/blob/master/pkg/proxy/proxy.go#L34-L40

Can we reuse the any of the proxy code here?

Reply:
Outdated after the switch to `go-client`, we now proxy no data in user settings handler anymore. Data are returned by `serverutils.SendResponse`